lib/collection/src/shards/replica_set/update.rs

```diff
@@ -228,20 +238,10 @@ impl ShardReplicaSet {
             }
         }
 
-        if !failures.is_empty() {
-            let write_consistency_factor = self
-                .collection_config
-                .read()
-                .await
-                .params
-                .write_consistency_factor
-                .get() as usize;
-            let minimal_success_count = write_consistency_factor.min(total_results);
-            if successes.len() < minimal_success_count {
-                // completely failed - report error to user
-                let (_peer_id, err) = failures.into_iter().next().expect("failures is not empty");
-                return Err(err);
-            }
+        if !failures.is_empty() && successes.len() < minimal_success_count {
```

Comment:
Could we add a comment before here to explain why we test these two conditions exactly?